hs/app/reach/Main.hs

```diff
@@ -2544,39 +2543,104 @@ support = command "support" $ info (pure step1) d
-      -- What if the user ignores instructions and hits 'y'
-      -- without actually authenticating, first?
-      let accessTokenPair = head $ filter isAccessTokenPair arrayOfArrayOfText
       -- @TODO: Save accessToken; git-credential-store
       -- Warning: Permission errors when doing this^
-      let accessToken = accessTokenPair !! 1
-      liftIO . T.putStrLn $ pack ""
-      -- @TODO: error handling!
-      -- What if index.mjs or index.rsh don't exist?
-      indexRsh <- liftIO $ readFile "index.rsh"
-      indexMjs <- liftIO $ readFile "index.mjs"
-      let indexRshValue =
-            object
-              [ "content" .= indexRsh
-              , "language" .= language
-              , "type" .= typeForUploadJson
-              ]
-      let indexMjsValue =
-            object
-              [ "content" .= indexMjs
-              , "language" .= language
-              , "type" .= typeForUploadJson
-              ]
-      let indexRshJson :: A.Pair
-          indexRshJson = ("index.rsh" .= indexRshValue)
-      let indexMjsJson :: A.Pair
-          indexMjsJson = ("index.mjs" .= indexMjsValue)
-      -- @TODO: Also add output of reach hashes!
-      let mainJson =
-            object
-              [ "files" .= object [ indexRshJson, indexMjsJson ]
-              ]
+      case headMay $ filter isAccessTokenPair arrayOfArrayOfText of
+        Nothing -> do
+          case headMay $ filter isErrorPair arrayOfArrayOfText of
+            Nothing -> liftIO . T.putStrLn $ pack
+                "Upload unsuccessful; couldn't find an authorization code, or an error!"
+            Just errorPair -> do
+              let reason = errorPair !! 1
+              liftIO . T.putStr $ pack "\nError while acquiring access token: "
+              liftIO . T.putStrLn $ reason
+        Just accessTokenPair -> do
+          let accessToken = accessTokenPair !! 1
+          completeStep3WithThe accessToken
+
+    completeStep3WithThe :: Text -> ReaderT Env IO ()
+    completeStep3WithThe accessToken = do
+      indexRshExists <- liftIO $ doesFileExist "index.rsh"
+      case indexRshExists of
+        True -> do
+          indexRsh <- liftIO $ readFile "index.rsh"
+          let indexRshValue =
+                object
+                  [ "content" .= indexRsh
+                  , "language" .= language
+                  , "type" .= typeForUploadJson
+                  ]
+          let indexRshJson :: A.Pair
+              indexRshJson = ("index.rsh" .= indexRshValue)
+          checkIndexMjs indexRshJson True accessToken
+        False -> do
+          liftIO . T.putStrLn $ pack ""
+          liftIO . T.putStrLn $ pack
+            "Didn't find index.rsh in the current directory; skipping..."
+          -- @HACK Trying to pass an "empty" A.Pair...
+          let nullString :: String
+              nullString = ""
+          let null' :: A.Pair
+              null' = ("empty" .= nullString)
+          checkIndexMjs null' False accessToken
+
```

Comment:
Shudder

Reply:
I don't know whether you dislike the hack to get an "empty" pair, the blank line, or both.

I'm going to assume you just don't like the blank line and remove it.